src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java

```diff
@@ -187,13 +188,13 @@ public static int findTapPosition(PhaseTapChanger ptc, double angle) {
     /**
      * Find the tap position of a ratio tap changer corresponding to a given rho shift.
      */
-    public static int findTapPosition(RatioTapChanger rtc, double ptcRho, double rho) {
+    public static OptionalInt findTapPosition(RatioTapChanger rtc, double ptcRho, double rho) {
         for (int position = rtc.getLowTapPosition(); position <= rtc.getHighTapPosition(); position++) {
             if (Math.abs(rho - ptcRho * rtc.getStep(position).getRho()) < EPS_ALPHA) {
-                return position;
+                return OptionalInt.of(position);
             }
         }
-        throw new PowsyblException("No tap position found (should never happen)");
```

Comment:
Do we need to log an error message or not? It is the issue I met last week but I am not sure to understand why we can have no tap position...